x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceMatcher.java

```diff
@@ -218,18 +239,88 @@ private void match(int stage, SequenceKey key, Ordinal ordinal, HitReference hit
                 }
             }
 
-            completed.add(sequence);
+            tryComplete(sequence);
             // update the bool lazily
             // only consider positive limits / negative ones imply tail which means having to go
             // through the whole page of results before selecting the last ones
             // doing a limit early returns the 'head' not 'tail'
-            headLimit = limit != null && limit.limit() > 0 && completed.size() == limit.totalLimit();
+            calculateHeadLimit();
         } else {
             stageToKeys.add(stage, key);
             keyToSequences.add(stage, sequence);
         }
     }
 
+    public void tryComplete(Sequence sequence) {
+        for (int i = 0; i < missingEvents.length; i++) {
+            if (missingEvents[i]) {
+                toCheckForMissing.add(sequence);
+                return;
+            }
+        }
```

Comment:
I read this as "if there is any missing event filter in this sequence, then check the current sequence". This could easily be a boolean (where if there is at least one `true` value in `missingEvents`), no need to recheck `missingEvents` in a `for` loop every time.